src/bootstrap/dist.rs

```diff
@@ -1973,7 +1982,11 @@ impl Step for RustDev {
         // `$ORIGIN/../lib` can find it. It may also be used as a dependency
         // of `rustc-dev` to support the inherited `-lLLVM` when using the
         // compiler libraries.
-        maybe_install_llvm(builder, target, &tarball.image_dir().join("lib"));
+        let dst_libdir = tarball.image_dir().join("lib");
+        if maybe_install_llvm(builder, target, &dst_libdir) {
+            let link_type = if builder.config.llvm_link_shared { "dynamic" } else { "static" };
+            t!(std::fs::write(dst_libdir.join("link-type.txt"), link_type), dst_libdir);
```

Comment:
I'd rather just always write this file and avoid the complexity of returning true/false from maybe_install - is it necessary?

Reply:
Yes, otherwise `lib/` won't exist and this will short circuit with an error.

Reply:
Can we write it elsewhere in the tarball to avoid this problem?

Reply:
I think so, yes. I pushed a commit writing it to the top-level of the tarball instead.

Reply:
Please squash it in - modulo my other concern, this PR looks good to me.